apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Footer/Menu/MenuActionButton/MenuActionButton.tsx

```diff
@@ -0,0 +1,227 @@
+import Button from '@mui/material/Button'
+import { useTranslation } from 'next-i18next'
+import { ReactElement } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+
+import { TreeBlock } from '@core/journeys/ui/block'
+import { useCommand } from '@core/journeys/ui/CommandProvider'
+import {
+  ActiveContent,
+  ActiveSlide,
+  useEditor
+} from '@core/journeys/ui/EditorProvider'
+import { useJourney } from '@core/journeys/ui/JourneyProvider'
+import ChevronRight from '@core/shared/ui/icons/ChevronRight'
+
+import {
+  BlockFields_CardBlock,
+  BlockFields_StepBlock
+} from '../../../../../../../../../__generated__/BlockFields'
+import {
+  ThemeMode,
+  ThemeName,
+  TypographyVariant
+} from '../../../../../../../../../__generated__/globalTypes'
+import { setBlockRestoreEditorState } from '../../../../../../utils/useBlockDeleteCommand'
+import { MENU_BLOCK_X, MENU_BLOCK_Y } from '../constants'
+
+import { useMenuBlockCreateMutation } from './useMenuBlockCreateMutation/useMenuBlockCreateMutation'
+import {
+  removeCachedBlocks,
+  useMenuBlockDeleteMutation
+} from './useMenuBlockDeleteMutation/useMenuBlockDeleteMutation'
+import { useMenuBlockRestoreMutation } from './useMenuBlockRestoreMutation/useMenuBlockRestoreMutation'
+
+export function MenuActionButton(): ReactElement {
+  const { t } = useTranslation('apps-journeys-admin')
+  const { journey } = useJourney()
+  const { add } = useCommand()
```

Comment:
Not sure if you've already tried this, but is it possible to use the existing command `add` and `delete`? It should save on some complexity you have to do for the undo/redo
```suggestion
const { addBlock } = useBlockCreateCommand()
const { addBlockDelete } = useBlockDeleteCommand()
```

Reply:
You might be able to do this if you need to do custom redo/undo mutations though